doc/specifications/proposed/replicator/architecture.rst

```diff
@@ -0,0 +1,142 @@
+==========
+Replicator
+==========
+
+This document describes a common synchronization mechanism between two systems,
+which can cope with write failures, network downtimes and other types of
+partitions. The approach only works with a single write master (Single Source
+of Truth) and any number of read slaves.
```

Comment:
TIP: design the system/api to be resilient to edits done on the slave as well. Maybe not as a guarantee that system will work, but this will happen in real life